openmc/data/ace.py

```diff
@@ -0,0 +1,460 @@
+
+            # Check for end-of-file
+            if len(ace_file.read(1)) == 0:
+                return
+            ace_file.seek(start_position)
+
+            # Read name, atomic mass ratio, temperature, date, comment, and
+            # material
+            name, atomic_weight_ratio, temperature, date, comment, mat = \
+                struct.unpack(str('=10sdd10s70s10s'), ace_file.read(116))
+            name = name.decode().strip()
+
+            # Read ZAID/awr combinations
+            data = struct.unpack(str('=' + 16*'id'), ace_file.read(192))
+            pairs = list(zip(data[::2], data[1::2]))
+
+            # Read NXS
+            nxs = list(struct.unpack(str('=16i'), ace_file.read(64)))
+
+            # Determine length of XSS and number of records
+            length = nxs[0]
+            n_records = (length + entries - 1)//entries
+
+            # verify that we are supposed to read this table in
+            if (table_names is not None) and (name not in table_names):
+                ace_file.seek(start_position + recl_length*(n_records + 1))
+                continue
+
+            if verbose:
+                kelvin = round(temperature * 1e6 / 8.617342e-5)
+                print("Loading nuclide {0} at {1} K".format(name, kelvin))
+
+            # Read JXS
+            jxs = list(struct.unpack(str('=32i'), ace_file.read(128)))
+
+            # Read XSS
+            ace_file.seek(start_position + recl_length)
+            xss = list(struct.unpack(str('={0}d'.format(length)),
+                                     ace_file.read(length*8)))
+
+            # Insert zeros at beginning of NXS, JXS, and XSS arrays so that the
+            # indexing will be the same as Fortran. This makes it easier to
+            # follow the ACE format specification.
+            nxs.insert(0, 0)
+            nxs = np.array(nxs, dtype=int)
+
+            jxs.insert(0, 0)
+            jxs = np.array(jxs, dtype=int)
+
+            xss.insert(0, 0.0)
+            xss = np.array(xss)
+
+            # Create ACE table with data read in
+            table = Table(name, atomic_weight_ratio, temperature, pairs,
+                          nxs, jxs, xss)
+            self.tables.append(table)
+
+            # Advance to next record
+            ace_file.seek(start_position + recl_length*(n_records + 1))
+
+    def _read_ascii(self, ace_file, table_names, verbose=False):
+        """Read an ASCII (Type 1) ACE table.
+
+        Parameters
+        ----------
+        ace_file : file
+            Open ACE file
+        table_names : None, str, or iterable
+            Tables from the file to read in.  If None, reads in all of the
+            tables. If str, reads in only the single table of a matching name.
+        verbose : str, optional
+            Whether to display what tables are being read. Defaults to False.
+
+        """
+
+        tables_seen = set()
+
+        lines = [ace_file.readline() for i in range(13)]
```

Comment:
13 seems like a magic number used here and elsewhere. Can we move it to a constant?